packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx

```diff
@@ -114,37 +115,124 @@ const StartVisitForm: React.FC<StartVisitFormProps> = ({
       {},
     );
 
-    return z.object({
-      visitStartDate: z.date(),
-      visitStartTime: z
-        .string()
-        .refine((value) => value.match(time12HourFormatRegex), t('invalidTimeFormat', 'Invalid time format')),
-      visitStartTimeFormat: z.enum(['PM', 'AM']),
-      visitStopDate: displayVisitStopDateTimeFields ? z.date() : z.date().optional(),
-      visitStopTime: displayVisitStopDateTimeFields
-        ? z
-            .string()
-            .refine((value) => value.match(time12HourFormatRegex), t('invalidTimeFormat', 'Invalid time format'))
-        : z.string().optional(),
-      visitStopTimeFormat: displayVisitStopDateTimeFields ? z.enum(['PM', 'AM']) : z.enum(['PM', 'AM']).optional(),
-      programType: z.string().optional(),
-      visitType: z.string().refine((value) => !!value, t('visitTypeRequired', 'Visit type is required')),
-      visitLocation: z.object({
-        display: z.string(),
-        uuid: z.string(),
-      }),
-      visitAttributes: z.object(visitAttributes),
-    });
-  }, [t, config, displayVisitStopDateTimeFields]);
+    let maxVisitStartDatetime: Date = null,
+      minVisitStopDatetime: Date = null;
+
+    if (visitToEdit?.encounters?.length) {
+      const allEncountersDateTime = visitToEdit?.encounters?.map(({ encounterDatetime }) =>
+        Date.parse(encounterDatetime),
+      );
+      maxVisitStartDatetime = new Date(Math.min(...allEncountersDateTime));
+      minVisitStopDatetime = new Date(Math.max(...allEncountersDateTime));
```

Comment:
Spread operators (`...`) seem like a mistake here. `Math.min()` and `Math.max()` do not alter the array, so there's no need to copy them here.

Reply:
Thanks for the catch!

Reply:
![image](https://github.com/openmrs/openmrs-esm-patient-chart/assets/51502471/a90dae8f-4f5e-4230-9106-5ea8bab93824)
It is the requirement of the `Math.min/max`